### PR TITLE
[POC] Use Travis CI GithubAction

### PR DESCRIPTION
**Pull request by OskarStark**

### CODE REVIEW

**Comment by TheGarious** [comment]
Look in this repository github : [Link](https://github.com/travis-ci/actions) 

You need added a token from Travis CI.

```
action "Test on Travis CI" {
  uses = "travis-ci/actions@master"
  secrets = [
    "TRAVIS_TOKEN",
  ]
}
```

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+on: [push, pull_request]
+name: Build
+jobs:
+    travis-ci:
+        name: Travis CI
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@master
+            - name: Travis CI
+              uses: travis-ci/actions@master
```
Comment (OskarStark) [inline .github/workflows/build.yaml]:
```suggestion
              uses: travis-ci/actions@master
              env:
                  TRAVIS_TOKEN: ${{ secrets.TRAVIS_TOKEN }}
```

**Comment by TheGarious** [comment]
Try this : 
```
uses = "travis-ci/actions@master"
secrets = [
    ${{ secrets.TRAVIS_TOKEN }},
]
```

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+              env:
```
Comment (OskarStark) [inline .github/workflows/build.yaml]:
```suggestion
              secrets = [
```

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+                  TRAVIS_TOKEN: ${{ secrets.TRAVIS_TOKEN }}
```
Comment (OskarStark) [inline .github/workflows/build.yaml]:
```suggestion
    ${{ secrets.TRAVIS_TOKEN }},
]
```

**Comment by TheGarious** [comment]
Can you try with this code 

```
workflow "Launch Travis" {
  on: [push, pull_request]
  resolves = ["travis-ci"]
}

action "travis-ci" {
  uses = "travis-ci/actions@master"
  secrets = [
      ${{ secrets.TRAVIS_TOKEN }},
  ]
}

```

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+name: Build
+jobs:
+    travis-ci:
+        name: Travis CI
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@master
+            - name: Travis CI
+              uses: travis-ci/actions@master
+              secrets = [
```
Comment (OskarStark) [inline .github/workflows/build.yaml]:
```suggestion
              secrets:
```

File: .github/workflows/build.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+                  ${{ secrets.TRAVIS_TOKEN }},
```
Comment (OskarStark) [inline .github/workflows/build.yaml]:
```suggestion
                  - ${{ secrets.TRAVIS_TOKEN }},
```

File: .github/workflows/build.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+              ]
```
Comment (OskarStark) [inline .github/workflows/build.yaml]:
```suggestion

```

Comment (TheGarious) [reply .github/workflows/build.yaml]: `secrets` replace by `secret`

Comment (OskarStark) [reply .github/workflows/build.yaml]: Not sure it’s right, because I don’t know if this is valid yaml anyway:

secret = [...]

Comment (TheGarious) [reply .github/workflows/build.yaml]: Github error say that error. I saw many article from github action with this syntax.

Comment (TheGarious) [reply .github/workflows/build.yaml]: If you want, tonight i try github action with my personal project affiliate travis ci for test.

**Comment by TheGarious** [comment]
With my [build.yaml ](https://github.com/TheGarious/demo-for-blog/blob/master/.github/workflows/build.yml) i have got access denied i need generate a token. But for this repo, i think is good.

File: .github/workflows/build.yaml (hunk 6 above)
Comment (OskarStark) [reply .github/workflows/build.yaml]: Yes please

Comment (TheGarious) [reply .github/workflows/build.yaml]: With my [build.yaml](With my build.yaml i have got access denied i need generate a token. But for this repo, i think is good.) i have got access denied i need generate a token. But for this repo, i think is good.

I think an article for create an access token for travis : [Link](https://medium.com/@JoshuaTheMiller/retrieving-your-travis-ci-api-access-token-bc706b2b625a)

I search today for find a solution

File: .github/workflows/build.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+jobs:
+    travis-ci:
+        name: Travis CI
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@master
+            - name: Travis CI
+              uses: travis-ci/actions@master
+              env:
+                  TRAVIS_TOKEN: ${{ secrets.SECRET_TRAVIS_CI }}
```
Comment (OskarStark) [inline .github/workflows/build.yaml]:
```suggestion
                  TRAVIS_TOKEN: ${{ secrets.TRAVIS_TOKEN }}
```

File: .github/workflows/build.yaml (hunk 3 above)
Comment (OskarStark) [inline .github/workflows/build.yaml]: @nicolas-grekas could you please set a secret for this repository with the symfony travis token?

Name: `TRAVIS_TOKEN`

Comment (nicolas-grekas) [reply .github/workflows/build.yaml]: might be good

Comment (TheGarious) [reply .github/workflows/build.yaml]: @OskarStark You can relaunch ci with actual commit or you need commit a changed for launch ci ?

Comment (OskarStark) [reply .github/workflows/build.yaml]: Thanks Nicolas

Will push force in a minute

**Comment by OskarStark** [comment]
I don't know why its not working :(

**Comment by TheGarious** [comment]
Again an Access Denied :/..

**Comment by TheGarious** [comment]
I send a message for support Travis. I investigate why we have got again this error.

**Comment by OskarStark** [comment]
Thank you

**Comment by javiereguiluz** [comment]
Oskar, is this still POC? What can we do to move it forward? thanks!

**Comment by OskarStark** [comment]
@TheGarious contacted @nicolas-grekas about a problem here, so yes its not working right now, but it should be 👍 

@TheGarious can you give Javier some more context here? Thank you 🙏

**Comment by TheGarious** [comment]
Hello @javiereguiluz  and @OskarStark,

I work in my private repository for find a solution for this problem. After that i give some instruction at @nicolas-grekas  for finish this POC.

Thanks

**Comment by OskarStark** [comment]
Thank you

**Comment by TheGarious** [comment]
Travis send me after few weeks ... multiple link for help me to resolve this problem.

https://github.com/BanzaiMan/travis-ci-action
https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets

I work this week-end for finish this implementation :+1:

**Comment by OskarStark** [comment]
Thanks 🙏

**Comment by TheGarious** [comment]
Hello @OskarStark , i try since few day to understand their documentation and can't test into my public repository. 

I sended an email at GithubAction, included a link into this pull request for finish this.

**Comment by OskarStark** [comment]
Any news on this @TheGarious ?

**Comment by TheGarious** [comment]
No, i send an email at github support, i can't now test github action into my personnal repository... If you want transfert a response from github support ?

**Pull request closed without merging**